tools/manifest/manifest.py

```diff
@@ -347,55 +387,79 @@ def update(self, tree):
                             del test_data[rel_path]
 
         if reftest_changes:
-            reftests, reftest_nodes, changed_hashes = self._compute_reftests(reftest_nodes)
-            data["reftest"].data = reftests
-            data["reftest_node"].data = reftest_nodes
+            reftests, reftest_nodes, changed_hashes = self._compute_reftests(all_reftest_nodes)
+            reftest_data = data["reftest"]
+            reftest_data.clear()
+            for path, items in iteritems(reftests):
+                if MYPY:
+                    reftest_data[path] = cast(Set[ManifestItem], items)
+                else:
+                    reftest_data[path] = items
+
+            reftest_node_data = data["reftest_node"]
+            reftest_node_data.clear()
+            for node_path, node_items in iteritems(reftest_nodes):
+                if MYPY:
+                    reftest_node_data[node_path] = cast(Set[ManifestItem], node_items)
+                else:
+                    reftest_node_data[node_path] = node_items
```

Comment:
@jgraham thoughts on this?

Reply:
Eh, making the control flow depend on whether we're running the type checker or not seems terrifying. What's the underlying requirement here?

Reply:
See the docs about the [invariance of `List`](https://mypy.readthedocs.io/en/latest/generics.html#variance-of-generic-types); the same applies to `Set`.

I guess the alternative in this case is to make `Manifest._compute_reftests` return `Set[ManifestItem]` rather than `Set[RefTest]`?